### PR TITLE
HakawaiSwiftDemo

### PR DESCRIPTION
**Pull request by sergstav**
Implemented swift demo for mentions plugin.

### CODE REVIEW

File: HakawaiDemoSwift/HakawaiDemoSwift.xcodeproj/xcuserdata/sergstavitskii.xcuserdatad/xcschemes/xcschememanagement.plist [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (jmkk) [inline HakawaiDemoSwift/HakawaiDemoSwift.xcodeproj/xcuserdata/sergstavitskii.xcuserdatad/xcschemes/xcschememanagement.plist]: Remove this file from the commit, it's your personal scheme, not a shared one.

File: HakawaiDemoSwift/Podfile.lock [hunk 2]
```diff
@@ -0,0 +1,17 @@
+PODS:
+  - Hakawai (5.1.1):
```
Comment (jmkk) [inline HakawaiDemoSwift/Podfile.lock]: Please update to use the latest Hakawai version

File: HakawaiDemoSwift/HakawaiDemoSwift/Mentions/MentionsDemoViewController.swift [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+import Hakawai
+
+class MentionsDemoViewController: UIViewController {
+    
+    //MARK: - Outlets
+    @IBOutlet weak var textView: HKWTextView!
+    @IBOutlet weak var mentionsListButton: UIButton!
+    
+    //
+    private var plugin: HKWMentionsPlugin!
```
Comment (jmkk) [inline HakawaiDemoSwift/HakawaiDemoSwift/Mentions/MentionsDemoViewController.swift]: Please review and apply LinkedIn's Swift Code Style - https://github.com/linkedin/swift-style-guide

For example here you should not use implicitly unwrapped optional, and make it a regular optional.

File: HakawaiDemoSwift/HakawaiDemoSwift/Mentions/MentionsEntity.swift [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+//  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+//  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//
+
+import Foundation
+import Hakawai
+
+class MentionsEntity: NSObject {
+
+    private var name: String!
```
Comment (jmkk) [inline HakawaiDemoSwift/HakawaiDemoSwift/Mentions/MentionsEntity.swift]: Any reason for those to be (implicitly unwrapped) optionals?

File: HakawaiDemoSwift/HakawaiDemoSwift/Mentions/MentionsDemoViewController.swift [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+        // If the text view loses focus while the mention chooser is up, and then regains focus, it will automatically put
+        //  the mentions chooser back up
+        mentionsPlugin?.resumeMentionsCreationEnabled = true
+        // Add edge insets so chooser view doesn't overlap the text view's cosmetic grey border
+        mentionsPlugin?.chooserViewEdgeInsets = UIEdgeInsetsMake(2, 0.5, 0.5, 0.5)
+        plugin = mentionsPlugin
+        plugin.chooserViewBackgroundColor = .lightGray
+        // The mentions plug-in requires a delegate, which provides it with mentions entities in response to a query string
+        mentionsPlugin?.delegate = MentionsManager.shared
+//        mentionsPlugin?.stateChangeDelegate =
```
Comment (jmkk) [inline HakawaiDemoSwift/HakawaiDemoSwift/Mentions/MentionsDemoViewController.swift]: Remove

File: HakawaiDemoSwift/HakawaiDemoSwift/Mentions/MentionsManager.swift [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+
+//MARK: - HKWMentionsDelegate
+extension MentionsManager: HKWMentionsDelegate {
+    
+    // In this method, the plug-in gives us a search string and some metadata, as well as a block. Our responsibility is to
+    //  perform whatever work is necessary to get the entities for that search string (network call, database query, etc),
+    //  and then to call the completion block with an array of entity objects corresponding to the search string. See the
+    //  documentation for the method for more details.
+    func asyncRetrieveEntities(forKeyString keyString: String!, searchType type: HKWMentionsSearchType, controlCharacter character: unichar, completion completionBlock: (([Any]?, Bool, Bool) -> Void)!) {
+        if completionBlock == nil {
```
Comment (jmkk) [inline HakawaiDemoSwift/HakawaiDemoSwift/Mentions/MentionsManager.swift]: Prefer `guard` semantics here.

File: HakawaiDemoSwift/HakawaiDemoSwift/Mentions/MentionsManager.swift [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+        }
+        completionBlock(buffer, false,  true)
+    }
+    
+    // In this method, the plug-in gives us a mentions entity (one we previously returned in response to a query), and asks
+    //  us to provide a table view cell corresponding to that entity to be presented to the user.
+    func cell(forMentionsEntity entity: HKWMentionsEntityProtocol!, withMatch matchString: String!, tableView: UITableView!) -> UITableViewCell! {
+        var cell = tableView.dequeueReusableCell(withIdentifier: "mentionsCell")
+        if cell == nil {
+            cell = UITableViewCell(style: .subtitle, reuseIdentifier: "mentionsCell")
```
Comment (jmkk) [inline HakawaiDemoSwift/HakawaiDemoSwift/Mentions/MentionsManager.swift]: Add and reuse a constant for the `mentionsCell` string literal.

**Comment by sergstav** [comment]
Updated code, please review.

**Pull request merged**